developer-extension/src/panel/sessionReplayPlayer/startSessionReplayPlayer.ts

```diff
@@ -50,20 +50,25 @@ export function startSessionReplayPlayer(
     }
   })
 
-  const stopListeningToSdkMessages = listenSdkMessages((message) => {
-    if (message.type === 'record') {
-      const record = message.payload.record
-      if (status === 'loading') {
-        bufferedRecords.add(record)
-      } else if (status === 'waiting-for-full-snapshot') {
-        if (isFullSnapshotStart(record)) {
-          status = 'ready'
-          onStatusChange(status)
-          messageBridge.sendRecord(record)
-        }
-      } else {
+  const backgroundMessageSubscription = onBackgroundMessage.subscribe((backgroundMessage) => {
+    if (backgroundMessage.type !== 'sdk-message') {
+      return
+    }
+    const sdkMessage = backgroundMessage.message
+    if (sdkMessage.type !== 'record') {
+      return
+    }
+    const record = sdkMessage.payload.record
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏simpler alternative
```suggestion
    if (!(backgroundMessage.type === 'sdk-message' && backgroundMessage.message.type === 'record')) {
      return
    }
    const record = backgroundMessage.message.payload.record
```